gem/spec/carthage_spec.rb

```diff
@@ -0,0 +1,48 @@
+#
+#  Copyright 2016 Google Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+require_relative 'spec_helper'
+
+describe 'carthage' do
+  NIL_YAML = "--- \n...\n".freeze
+
+  it 'successfully modifies the xcode project' do
+    xcodeproj_2 = File.expand_path(File.join(__dir__, 'fixtures', 'carthage_after', 'Example.xcodeproj'))
+
+    Dir.mktmpdir do |tmp_dir|
+      FileUtils.cp_r carthage_before, tmp_dir
+
+      Dir.chdir tmp_dir do
+        # carthage modification of xcodeproj is non-deterministic so we can't rely on
```

Comment:
Sorry if I'm missing something here, but couldn't we search for specific things like files added or changes in the project file instead of having two projects present for testing? This increases the size of the EarlGrey repo.

Instead of diffing, maybe we could have a script that checked for changes to the Scheme, if EarlGrey was added to the Build Phases, Framework Search Paths, a Build Phases scrip was correct and so on.